pkg/store/consul/consulutil/watch_test.go

```diff
@@ -207,6 +208,65 @@ func TestWatchPrefix(t *testing.T) {
 	}
 }
 
+// TestWatchKeys verifies some basic operations of the WatchKeys() function. It
+// should find existing keys, send new updates when the data changes, and
+// ignore changes outside its prefix.
+func TestWatchKeys(t *testing.T) {
+	t.Parallel()
+	f := NewFixture(t)
+	defer f.Stop()
+
+	done := make(chan struct{})
+	defer func() {
+		if done != nil {
+			close(done)
+		}
+	}()
+	kv1a := &api.KVPair{Key: "prefix/hello", Value: []byte("these")}
+	kv1b := &api.KVPair{Key: "prefix/hello", Value: []byte("do")}
+	kv2a := &api.KVPair{Key: "prefix/test", Value: []byte("not")}
+	kv3a := &api.KVPair{Key: "something", Value: []byte("matter")}
+
+	// Process existing data
+	f.Client.KV().Put(kv1a, nil)
+	keysChan := WatchKeys("prefix/", f.Client.KV(), done, 0)
+	keys := <-keysChan
+	if !reflect.DeepEqual(keys.Keys, []string{kv1a.Key}) {
+		t.Error("existing data not recognized")
+	}
+
+	// Get an updates when the data changes (create, modify, delete)
+	f.Client.KV().Put(kv1b, nil)
+	keys = <-keysChan
+	if !reflect.DeepEqual(keys.Keys, []string{kv1b.Key}) {
+		t.Error("keys changed inappropriately")
+	}
+
+	f.Client.KV().Put(kv2a, nil)
+	keys = <-keysChan
+	if !reflect.DeepEqual(keys.Keys, []string{kv1a.Key, kv2a.Key}) {
+		t.Error("did not find new key")
+	}
+
+	f.Client.KV().Delete(kv1a.Key, nil)
+	keys = <-keysChan
+	if !reflect.DeepEqual(keys.Keys, []string{kv2a.Key}) {
+		t.Error("did not notice key deletion")
+	}
+
+	// The watcher should ignore kv3a, which is outside its prefix
+	f.Client.KV().Put(kv3a, nil)
+	f.Client.KV().Delete(kv2a.Key, nil)
+	keys = <-keysChan
+	if len(keys.Keys) != 0 {
+		t.Error("watch did not ignore keys outside of prefix")
+	}
+	close(done)
+	done = nil
+	for _ = range keysChan {
```

Comment:
just `for range keysChan` will suffice... some tool will correct us, though maybe we don't run it by default. Maybe it's `go fmt -s` or something.